apps/client/src/app/fhir-edit/multi-jurisdiction/multi-jurisdiction.component.ts

```diff
@@ -51,28 +57,104 @@ export class FhirMultiJurisdictionComponent implements OnInit {
     }
   }
 
-  addCoding(jurisdiction: ICodeableConcept) {
+  addCoding(jurisdiction: ICodeableConcept, index: number) {
     jurisdiction.coding = jurisdiction.coding || [];
     jurisdiction.coding.push({});
+    if(this.editFields.length < index + 1){
+      this.editFields.push([]);
+    }
+    this.editFields[index].push(false);
   }
 
   isInvalid(){
-    return this.required && (!this.jurisdictions[0].text ||
-      (this.jurisdictions[0].coding && (!this.jurisdictions[0].coding[0].code || !this.jurisdictions[0].coding[0].display || !this.jurisdictions[0].coding[0].system)));
+    return this.required && (this.jurisdictions && this.checkForCompletedJurisdiction());
+  }
+
+  checkForCompletedJurisdiction(){
+    for(let x = 0; x < this.jurisdictions.length; x++){
+      if(this.jurisdictions[x].text && this.jurisdictions[x].coding){
+        for(let y = 0; y < this.jurisdictions[x].coding.length; y++){
+          if(this.jurisdictions[x].text && this.jurisdictions[x].coding[y].code && this.jurisdictions[x].coding[y].display
```

Comment:
display is typically optional. consider removing `&& this.jurisdictions[x].coding[y].display` from this if-check.

Reply:
I tend to agree about removing it, and considered leaving it as is, but was torn about what the user would see when they swapped from manual input to the dropdown after changing the code from say "US" to "UA".  Since "UA" is a code in the dropdown, the dropdown will now show "Ukraine (UA)". But, if you swap back and forth between manual and auto fill, the displays will be mismatched "United States of America" on the manual side but "Ukraine" on the auto side.  If you think that's an okay outcome, I can remove this part of the logic.  Please let me know.

Reply:
Nah, you're probably right. Go ahead and leave it.